src/main/java/org/apache/xml/security/algorithms/implementations/SignatureBaseRSA.java

```diff
@@ -62,7 +62,7 @@ public SignatureBaseRSA(Provider provider) throws XMLSignatureException {
         String algorithmID = JCEMapper.translateURItoJCEID(this.engineGetURI());
         this.signatureAlgorithm = getSignature(provider, algorithmID);
         LOG.log(Level.DEBUG, "Created SignatureRSA using {0} and provider {1}",
-            algorithmID, signatureAlgorithm.getProvider());
+            algorithmID, provider);
```

Comment:
Good catch, I understand the problem, sorry for not detecting this when I fixed this.

I would just restore the prior code and do not include the provider in the log message, because provider can be null and in that case doesn't really make sense for the log message as I was trying to log the provider that was chosen. So change back to:

`LOG.debug("Created SignatureRSA using {}", algorithmID);`

Reply:
Thanks, reverted debug line